src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1422,6 +1440,30 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                         }
                     }
 
+                    if (pendingAttachmentsTombstoneUpdates != null)
+                    {
+                        foreach (var (docId, cv, modifiedTicks) in pendingAttachmentsTombstoneUpdates)
+                        {
+                            var doc = context.DocumentDatabase.DocumentsStorage.Get(context, docId, DocumentFields.ChangeVector, throwOnConflict: false);
+                            
+                            // RavenDB-19421: if the document doesn't exist, the tombstone doesn't matter
+                            // and if the change vector is already merged, then it is already taken into consideration
+                            // we need to force an update when this is _not_ the case, because this replication batch gave us the tombstone only, without
+                            // the related document update, so we need to simulate that locally
+                            if (doc != null &&
+                                ChangeVectorUtils.GetConflictStatus(cv, doc.ChangeVector) != ConflictStatus.AlreadyMerged) 
+                            {   
+                                // have to load the full document
+                                doc = context.DocumentDatabase.DocumentsStorage.Get(context, docId, fields: DocumentFields.All, throwOnConflict: false);
+                                long lastModifiedTicks = Math.Max(modifiedTicks, doc.LastModified.Ticks); // old versions may send with 0 in the tombstone ticks
```

Comment:
the `LastModified` is a new addition to the schema?

Reply:
if we have a conflict should we generate a new `ticks` ?

Reply:
No, that is something we always had, we just never sent it over the wire.
The `Math.Max` there is to handle the older versions.

This isn't working with conflicts, since then `doc` would be null, and the operation would fix itself on conflict resolve

Reply:
What will happen if one of the nodes will not send it? asking about Interversion compatibility

Reply:
Then we will default to the last modified on the doc, which is still okay (will resolve the conflict to the other version when it arrives)

We are already sending `0` over the wire, so this is fine

Reply:
They are already sending 0